src/sources/journald.rs

```diff
@@ -287,12 +298,21 @@ impl JournalSource for Journalctl {
         }
 
         let mut child = command.spawn().context(JournalctlSpawn)?;
-        let stdout = child.stdout.take().unwrap();
-        let stdout = BufReader::new(stdout);
+        let mut stdout = BufReader::new(child.stdout.take().unwrap());
+        let jstream = Box::pin(async_stream::stream! {
```

Comment:
How about using `lines`? https://docs.rs/futures/0.3.5/futures/io/trait.AsyncBufReadExt.html#method.lines

```diff
diff --git a/src/sources/journald.rs b/src/sources/journald.rs
index a0d83948f..3115e7d66 100644
--- a/src/sources/journald.rs
+++ b/src/sources/journald.rs
@@ -298,17 +298,8 @@ impl JournalSource for Journalctl {
         }

         let mut child = command.spawn().context(JournalctlSpawn)?;
-        let mut stdout = BufReader::new(child.stdout.take().unwrap());
-        let jstream = Box::pin(async_stream::stream! {
-            loop {
-                let mut line = Vec::new();
-                yield match stdout.read_until(b'\n', &mut line).await {
-                    Ok(0) => break,
-                    Ok(_) => Ok::<String, io::Error>(String::from_utf8_lossy(&line).into()),
-                    Err(err) => Err::<String, io::Error>(err),
-                };
-            }
-        });
+        let stdout = BufReader::new(child.stdout.take().unwrap());
+        let jstream = stdout.lines().boxed();

         let pid = Pid::from_raw(child.id() as i32);
         Ok((
```

Reply:
It is not the same as `from_utf8_lossy`, though.

Reply:
Oh, I did not noticed that `lines` exists... but yeah there `from_utf8` used instead `from_utf8_lossy` =\

Reply:
hmm we can use `split(b'\n')`, https://docs.rs/tokio/0.2.22/tokio/io/trait.AsyncBufReadExt.html#method.split

Reply:
Great! I'll change it now.